File: 프로그래머스/lv3/42884. 단속카메라/단속카메라.py
def solution(routes):
    answer = 0
    routes = sorted(routes, key = lambda x: x[1])
    camera = -30001
    for s, e in routes:
        if not s <= camera <= e:
            answer += 1
            camera = e
        # checkable = False
        # for camera in cameras:
        #     if s <= camera[0] <= e or s <= camera[1] <= e:
        #         checkable = True
        # if not checkable:
        #     cameras.append((s, e))
    print(routes, camera)
    return answer

File: 프로그래머스/3/42884. 단속카메라/단속카메라.js
const compareNumber = (a, b) => a[1] - b[1]
function solution(routes) {
    let answer = 0;
    let lastCamera = -1;
    const newRoutes = routes.map(route => [route[0]+30000, route[1]+30000]).sort(compareNumber)
    for (let i = 0; i < newRoutes.length; i++) {
        const [start, end] = newRoutes[i]
        if (start > lastCamera || end < lastCamera) {
            answer += 1
            lastCamera = end
        }
    }
    return answer;
}